routes/api/contacts.js

```diff
@@ -1,25 +1,167 @@
 const express = require('express')
-
 const router = express.Router()
+const {Contacts} = require ("../schema")
+const { validateJoi } = require('./validation')
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
+router.get('/', async (req, res) => {
+   try{
+    const contacts = await Contacts.find()
+      res.json ({
+      status: "success",
+      code: 200,
+      data: contacts
+    })
+  } catch (error) {
+    res.status(400).json({
+      status: "error",
+      code: 400,
+      message: "Bad request"
+    })
+  }
 })
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
+router.get('/:contactId', async (req, res) => {
+  const {contactId} = req.params;
+  try {
+    const contact = await Contacts.findById(contactId)
+    res.json ({
+      status: "success",
+      code: 200,
+      data: contact,
+      message: "Your contact has been found"
+    })
+  }   catch (error) {
+    res.status(400).json({
+      status: "error",
+      code: 400,
+      message: "Bad request"
+    })
+  }
+  
 })
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
+router.post('/', async (req, res) => {
+  const body = req.body;
+  const validate = validateJoi(body);
+  
+  if (validate.error) {
+    return res.status(400).json({
+      status: "error",
+      code: 400,
+      message: "Validation failed", 
+      details: validate.error.details 
+    });
+  }
+
+  try {
+    const contact = new Contacts(validate.value);
+    await contact.save();
+    res.json({
+      status: "success",
+      code: 201,
+      data: contact,
+      message: "You just added a new contact",
+    });
+  } catch (error) {
+    res.status(400).json({
+      status: "error",
+      code: 400,
+      message: "Bad Request",
+    });
+  }
 })
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
+
+router.delete('/:contactId', async (req, res) => {
+  const { contactId } = req.params;
+  
+  try {
+   const deleteContact = await Contacts.deleteOne({_id: contactId})
+    res.json({
+      status: "success",
+      code: 200,
+      data: contact,
+      message: "You deleted contact from your list",
+    });
+  } catch (error) {
+    res.status(400).json({
+      status: "error",
+      code: 400,
+      message: "Bad Request",
+    });
+  }
```

Comment:
```suggestion
  } catch (error) {
  console.log(error)
    res.status(400).json({
      status: "error",
      code: 400,
      message: "Bad Request",
    });
  }
```
Dodaj sobie console logi aby widzieć w terminalu pełną wiadomość o błędzie.